src/Microsoft.OData.Core/UriParser/Parsers/ODataPathParser.cs

```diff
@@ -360,9 +376,47 @@ private bool TryHandleAsKeySegment(string segmentText)
             KeySegment previousKeySegment = this.FindPreviousKeySegment();
 
             KeySegment keySegment;
+
+            bool trybindingEscapeFunction = false;
+            IEdmFunction function = null;
+
+            if (segmentText[segmentText.Length - 1] == ':' && previous.EdmType != null)
+            {
+
+                IEdmType bindingType = segmentText.Length == 1 ? previous.EdmType : previous.EdmType.AsElementType();

```

Comment:
>IEdmType bindingType = segmentText.Length == 1 ? previous.EdmType : previous.EdmType.AsElementType(); [](start = 16, length = 101)

the logic for this line seems obscure; please describe in a comment.

this says that, if the segment contains only a colon, and the previous segment was a collection, return the collection type; otherwise return the element type.  Right? What situation is this trying to handle?